lib/chartjs.dart

```diff
@@ -501,6 +501,21 @@ abstract class ChartTooltipCallback {
       List<ChartTooltipItem> item, ChartData data);
   external dynamic /*String|List<String>*/ afterFooter(
       List<ChartTooltipItem> item, ChartData data);
+  external factory ChartTooltipCallback({
+    beforeTitle: dynamic,
+    title: dynamic,
+    afterTitle: dynamic,
+    beforeBody: dynamic,
+    beforeLabel: dynamic,
+    label: dynamic,
+    labelColor: dynamic,
+    labelTextColor: dynamic,
+    afterLabel: dynamic,
+    afterBody: dynamic,
+    beforeFooter: dynamic,
+    footer: dynamic,
+    afterFooter: dynamic
+  });
```

Comment:
Can we try to specify the types, even for functions? I think it'll be helpful if using the constructor not to have to refer to the class member for the type.

I think this should work, but I haven't tested it. If you already have some code against it, could you give it a try, otherwise plmk and I'll mock up something for a test.

```suggestion
  external factory ChartTooltipCallback({
      dynamic /*String|List<String>*/ Function(
          List<ChartTooltipItem> item, ChartData data) beforeTitle,
      dynamic /*String|List<String>*/ Function(
          List<ChartTooltipItem> item, ChartData data) title,
      dynamic /*String|List<String>*/ Function(
          List<ChartTooltipItem> item, ChartData data) afterTitle,
      dynamic /*String|List<String>*/ Function(
          List<ChartTooltipItem> item, ChartData data) beforeBody,
      dynamic /*String|List<String>*/ Function(
          ChartTooltipItem tooltipItem, ChartData data) beforeLabel,
      dynamic /*String|List<String>*/ Function(
          ChartTooltipItem tooltipItem, ChartData data) label,
      ChartTooltipLabelColor Function(
          ChartTooltipItem tooltipItem, Chart chart) labelColor,
      String Function(
          ChartTooltipItem tooltipItem, Chart chart) labelTextColor,
      dynamic /*String|List<String>*/ Function(
          ChartTooltipItem tooltipItem, ChartData data) afterLabel,
      dynamic /*String|List<String>*/ Function(
          List<ChartTooltipItem> item, ChartData data) afterBody,
      dynamic /*String|List<String>*/ Function(
          List<ChartTooltipItem> item, ChartData data) beforeFooter,
      dynamic /*String|List<String>*/ Function(
          List<ChartTooltipItem> item, ChartData data) footer,
      dynamic /*String|List<String>*/ Function(
          List<ChartTooltipItem> item, ChartData data) afterFooter
  });
```

btw - the structure `{<arg>: dynamic, ...}` is different to `{dynamic <arg>, ...}`, which is how all the other constructors in this file are defined.

`dynamic` in recent versions of Dart is not just a type, but also a valid value (!) (albeit a weird one) which means that:
1. `<arg>: dynamic` defines an argument of type `dynamic` (by account of not having specifying the type explicitly), and initialises it with the value `dynamic`, whereas
2. `dynamic <arg>` defines an argument of type `dynamic` (as explicitly specified), and initialises with the value `null` (by account of not having specified an initial value).

So for these constructors, I think the expectation from the dart:js library is to define them of whatever type, but leave them un-initialised (ie. use the second version).

This is all quite whacky, so don't worry if it doesn't make immediate sense, just let me know if you need me to explain further etc.  Hopefully you won't need to use this very much, I know I don't want to!

Reply:
The suggestion code works fine. 

This makes sense - I didn't realise the error was that I had skipped the return type of the function. So I fell back to just using the return type dynamic which didn't seem right. 

Thanks for the explanation, read up a bit on `dynamic` - i think I have a better picture. Will note this.